ui-tests/src/test/java/org/openmrs/reference/TransferToWardServiceTest.java

```diff
@@ -11,42 +11,39 @@
 
 import org.junit.After;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
-import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
+import org.openmrs.reference.page.FindPatientPage;
+import org.openmrs.reference.page.PatientVisitsDashboardPage;
 import org.openmrs.uitestframework.test.TestData;
 
 import static org.junit.Assert.assertTrue;
 
-
 public class TransferToWardServiceTest extends ReferenceApplicationTestBase {
 
     private static final String INPATIENT_WARD = "Inpatient Ward";
-
     private static final String ISOLATION_WARD = "Isolation Ward";
-
     private TestData.PatientInfo testPatient;
 
     @Before
     public void createTestData() throws Exception {
         testPatient = createTestPatient();
-        String visitType = TestData.getAVisitType();
-        String visit = new TestData.TestVisit(testPatient.uuid, visitType, getLocationUuid(homePage)).create();
     }
 
-    @Ignore("TODO fix, inpatient_ward doesn't exisit on the page, fix workflow")
     @Test
-    public void transferToWardServiceTest() throws Exception {
-
-        ClinicianFacingPatientDashboardPage patientDashboardPage = homePage.goToActiveVisitPatient();
+    @Category(BuildTests.class)
+    public void transferToWardServiceTest() {
+        FindPatientPage findPatientPage = homePage.goToFindPatientRecord();
+        findPatientPage.enterPatient(testPatient.identifier);
+        findPatientPage.waitForPageToLoad();
+        PatientVisitsDashboardPage patientVisitsDashboardPage = findPatientPage.clickOnFirstPatient().startVisit();
 
-        patientDashboardPage.goToAdmitToInpatient().confirm(INPATIENT_WARD);
-        patientDashboardPage.waitForPage();
+        patientVisitsDashboardPage.goToAdmitToInpatient().confirm(INPATIENT_WARD);
 
-        patientDashboardPage.goToTransferToWardServicePage().confirm(ISOLATION_WARD);
-        patientDashboardPage.waitForPage();
+        patientVisitsDashboardPage.goToTransferToWardServicePage().confirm(ISOLATION_WARD);
 
-        assertTrue(patientDashboardPage.containsText("Outpatient"));
+        assertTrue(patientVisitsDashboardPage.containsText("Discharge"));
```

Comment:
Is there any reason why Outpatient would be disappearing?

Reply:
> Is there any reason why Outpatient would be disappearing?

The line code above assertTrue finds the test when its on TransferToWardServicePage and the visible text present is `Discharge` Not `Outpatient`

Reply:
I think rather i would at this moment confirm the existence of the 2 above encounters for Transfer as latest and Admission as first.

Start by waiting for the patientDashboard page and instead of validate the encounters

Reply:
Have handled this smartly within the commit with id `550f542`